stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java

```diff
@@ -386,12 +386,25 @@ private void runInternal() throws Exception {
       }
       CommonUtils.sleepMs(waitMs);
 
-      while (!Thread.currentThread().isInterrupted()
-          && ((!useStopCount && CommonUtils.getCurrentMs() < mContext.getEndMs())
-              || (useStopCount && mContext.mCounter.get() < mParameters.mStopCount))) {
+      long localCounter = 0;
+      while (true) {
+        if (Thread.currentThread().isInterrupted()) {
+          break;
+        }
+        if (!useStopCount) {
+          if (CommonUtils.getCurrentMs() >= mContext.getEndMs()) {
+            break;
+          }
+        } else {
+          localCounter = mContext.getCounter().getAndIncrement();
+          if (localCounter >= mParameters.mStopCount) {
+            break;
+          }
+        }
```

Comment:
```java
if (cond1) {
  break;
}

if (cond2) {
  break;
}

...

do your job
```

Reply:
Fixed in ae812ddc680ca967a61d7c5a4699eea4f7361fe7